src/test/java/com/training/services/tests/ProductsTest.java

```diff
@@ -62,8 +68,84 @@ public void testShorexProductTypeValidate() {
+
+        Map<String, Integer> dayMonthDaysInMonthFromLink = getDayMonthDaysInMonth(map.get("sailingIdParamValue").split("AL")[1]);
+
+        List<Products> products = response.getPayload().getProducts();
+        List<Offering> offerings;
+
+        for (int index = 0; index < products.size(); index++) {
+             offerings = products.get(index).getOffering();
+            for (int getDateOffering = 0; getDateOffering < offerings.size(); getDateOffering++) {
+
+                Map<String, Integer> dayMonthDaysInMonth = getDayMonthDaysInMonth(offerings.get(getDateOffering).getOfferingDate());
+
+                /**
+                 * If query paramter link date and offeringDate are from same month
+                 */
+                if (dayMonthDaysInMonthFromLink.get("month") == dayMonthDaysInMonth.get("month")) {
+                    if (!((dayMonthDaysInMonth.get("day") - dayMonthDaysInMonthFromLink.get("day")) <= 10)) {
+                        softAssert.fail("Product ID = " + offerings.get(getDateOffering)
+                                .getProductID() + " offering date is greater than 10 days = "
+                                + offerings.get(getDateOffering).getOfferingDate());
+                    }
+                } else {
+                    softAssert.assertThat(
+                            (dayMonthDaysInMonthFromLink.get("daysInMonth") - dayMonthDaysInMonthFromLink.get("day")) + dayMonthDaysInMonth.get("day") <= 10)
+                            .as("Product ID = "
+                                    + offerings.get(getDateOffering)
+                                    .getProductID() + " offeringDate is greater than 10 days = "
+                                    + offerings.get(getDateOffering)
+                                    .getOfferingDate())
+                            .isTrue();
+                }
+            }
+        }
+    }
+
+    /**
+     * Returning segregate day, month and number of days in a month
+     *
+     * @param actualDateFromResponse
+     * @return
+     * @throws ParseException
+     */
+    public Map<String, Integer> getDayMonthDaysInMonth(String actualDateFromResponse) throws ParseException {
+        Map<String, Integer> dateMap = new HashMap<>();
+
+        String dateSplit[] = new SimpleDateFormat("yyyy-MM-dd")
+                            .format(new SimpleDateFormat("yyyyMMdd")
+                            .parse(actualDateFromResponse))
+                            .split("-");
+
+        int getYear = Integer.parseInt(dateSplit[0]);
+        int getMonth = Integer.parseInt(dateSplit[1]);
+        int getDay = Integer.parseInt(dateSplit[2]);
+
+        /**
+         * Number of days in month
+         */
+        YearMonth yearMonthObject = YearMonth.of(getYear, getMonth);
+        int daysInMonth = yearMonthObject.lengthOfMonth();
+
+        dateMap.put("day", getDay);
+        dateMap.put("month", getMonth);
+        dateMap.put("daysInMonth", daysInMonth);
```

Comment:
Lets store the keys : day, month, daysInMonth in a global final String, and then use it everywhere in our code